Matchup business logic implementation

shaked8634 opened the pull request.

pranavsuri4303 committed:
fix up equipment service.

Commit:
Changes:

1.Implemented MatchUp repo, service, resolver.
2. Naming alignments
3. Added sample query

pranavsuri4303 committed:
Equipment Service Fixes

pranavsuri4303 committed:
Remove API Key

pranavsuri4303 committed:
Update bson tags and Equipment Service

pranavsuri4303 committed:
Remove Api key

pranavsuri4303 committed:
Federation for equipment

pranavsuri4303 committed:
remove unnecessary queries

pranavsuri4303 committed:
Add shared folders

diff --git a/equipment-service/graph/generated.go b/equipment-service/graph/generated.go
index f43d9ff..a2cf598 100644
--- a/equipment-service/graph/generated.go
+++ b/equipment-service/graph/generated.go
@@ -1658,6 +1658,8 @@ func (ec *executionContext) fieldContext_Entity_findTennisRacketByID(ctx context
 				return ec.fieldContext_TennisRacket_modelId(ctx, field)
 			case "weight":
 				return ec.fieldContext_TennisRacket_weight(ctx, field)
+			case "visibility":
+				return ec.fieldContext_TennisRacket_visibility(ctx, field)
 			}
 			return nil, fmt.Errorf("no field named %q was found under type TennisRacket", field.Name)
 		},
@@ -1891,6 +1893,8 @@ func (ec *executionContext) fieldContext_Mutation_createTennisRacket(ctx context
 				return ec.fieldContext_TennisRacket_modelId(ctx, field)
 			case "weight":
 				return ec.fieldContext_TennisRacket_weight(ctx, field)
+			case "visibility":
+				return ec.fieldContext_TennisRacket_visibility(ctx, field)
 			}
 			return nil, fmt.Errorf("no field named %q was found under type TennisRacket", field.Name)
 		},
@@ -1974,6 +1978,8 @@ func (ec *executionContext) fieldContext_Mutation_updateMyTennisRacket(ctx conte
 				return ec.fieldContext_TennisRacket_modelId(ctx, field)
 			case "weight":
 				return ec.fieldContext_TennisRacket_weight(ctx, field)
+			case "visibility":
+				return ec.fieldContext_TennisRacket_visibility(ctx, field)
 			}
 			return nil, fmt.Errorf("no field named %q was found under type TennisRacket", field.Name)
 		},
@@ -2425,6 +2431,8 @@ func (ec *executionContext) fieldContext_Query_myTennisRacket(ctx context.Contex
 				return ec.fieldContext_TennisRacket_modelId(ctx, field)
 			case "weight":
 				return ec.fieldContext_TennisRacket_weight(ctx, field)
+			case "visibility":
+				return ec.fieldContext_TennisRacket_visibility(ctx, field)
 			}
 			return nil, fmt.Errorf("no field named %q was found under type TennisRacket", field.Name)
 		},
@@ -3534,6 +3542,47 @@ func (ec *executionContext) fieldContext_TennisRacket_weight(_ context.Context,
 	return fc, nil
 }

+func (ec *executionContext) _TennisRacket_visibility(ctx context.Context, field graphql.CollectedField, obj *model.TennisRacket) (ret graphql.Marshaler) {
+	fc, err := ec.fieldContext_TennisRacket_visibility(ctx, field)
+	if err != nil {
+		return graphql.Null
+	}
+	ctx = graphql.WithFieldContext(ctx, fc)
+	defer func() {
+		if r := recover(); r != nil {
+			ec.Error(ctx, ec.Recover(ctx, r))
+			ret = graphql.Null
+		}
+	}()
+	resTmp, err := ec.ResolverMiddleware(ctx, func(rctx context.Context) (interface{}, error) {
+		ctx = rctx // use context from middleware stack in children
+		return obj.Visibility, nil
+	})
+	if err != nil {
+		ec.Error(ctx, err)
+		return graphql.Null
+	}
+	if resTmp == nil {
+		return graphql.Null
+	}
+	res := resTmp.(*model.Visibility)
+	fc.Result = res
+	return ec.marshalOVisibility2ᚖgithubᚗcomᚋCourtIQᚋcourtiqᚑbackendᚋequipmentᚑserviceᚋgraphᚋmodelᚐVisibility(ctx, field.Selections, res)
+}
+
+func (ec *executionContext) fieldContext_TennisRacket_visibility(_ context.Context, field graphql.CollectedField) (fc *graphql.FieldContext, err error) {
+	fc = &graphql.FieldContext{
+		Object:     "TennisRacket",
+		Field:      field,
+		IsMethod:   false,
+		IsResolver: false,
+		Child: func(ctx context.Context, field graphql.CollectedField) (*graphql.FieldContext, error) {
+			return nil, errors.New("field of type Visibility does not have child fields")
+		},
+	}
+	return fc, nil
+}
+
 func (ec *executionContext) _TennisString_id(ctx context.Context, field graphql.CollectedField, obj *model.TennisString) (ret graphql.Marshaler) {
 	fc, err := ec.fieldContext_TennisString_id(ctx, field)
 	if err != nil {
@@ -4282,6 +4331,8 @@ func (ec *executionContext) fieldContext_User_myTennisRackets(ctx context.Contex
 				return ec.fieldContext_TennisRacket_modelId(ctx, field)
 			case "weight":
 				return ec.fieldContext_TennisRacket_weight(ctx, field)
+			case "visibility":
+				return ec.fieldContext_TennisRacket_visibility(ctx, field)
 			}
 			return nil, fmt.Errorf("no field named %q was found under type TennisRacket", field.Name)
 		},
@@ -7090,6 +7141,8 @@ func (ec *executionContext) _TennisRacket(ctx context.Context, sel ast.Selection
 			out.Values[i] = ec._TennisRacket_modelId(ctx, field, obj)
 		case "weight":
 			out.Values[i] = ec._TennisRacket_weight(ctx, field, obj)
+		case "visibility":
+			out.Values[i] = ec._TennisRacket_visibility(ctx, field, obj)
 		default:
 			panic("unknown field " + strconv.Quote(field.Name))
 		}
diff --git a/equipment-service/graph/model/models_gen.go b/equipment-service/graph/model/models_gen.go
index bf97f3f..f89c0a2 100644
--- a/equipment-service/graph/model/models_gen.go
+++ b/equipment-service/graph/model/models_gen.go
@@ -75,6 +75,7 @@ type TennisRacket struct {
 	Model           *string             `json:"model,omitempty" bson:"model,omitempty"`
 	ModelID         *int                `json:"modelId,omitempty" bson:"modelId,omitempty"`
 	Weight          *float64            `json:"weight,omitempty" bson:"weight,omitempty"`
+	Visibility      *Visibility         `json:"visibility,omitempty" bson:"visibility,omitempty"`
 }

 func (TennisRacket) IsEquipment()                        {}
diff --git a/equipment-service/graph/schema/types/TennisRacket.gql b/equipment-service/graph/schema/types/TennisRacket.gql
index 644c46e..6376ccb 100644
--- a/equipment-service/graph/schema/types/TennisRacket.gql
+++ b/equipment-service/graph/schema/types/TennisRacket.gql
@@ -16,6 +16,7 @@ type TennisRacket implements Equipment @key(fields: "id") {
     model: String                         # Model of the racket (e.g., Blade 98)
     modelId: Int                          # Model ID for the racket (optional)
     weight: Float                         # Weight of the racket in grams (e.g., 305.0)
+    visibility: Visibility                # Visibility of the racket (PUBLIC, PRIVATE, FRIENDS)
 }

 extend type User @key(fields: "id") {
diff --git a/shared/graph/scalar/objectid.go b/shared/graph/scalar/objectid.go
new file mode 100644
index 0000000..8fcaf0f
--- /dev/null
+++ b/shared/graph/scalar/objectid.go
@@ -0,0 +1,31 @@
+package scalar
+
+import (
+	"fmt"
+	"io"
+
+	"github.com/99designs/gqlgen/graphql"
+	"go.mongodb.org/mongo-driver/bson/primitive"
+)
+
+// MarshalObjectID converts MongoDB ObjectID to GraphQL string
+func MarshalObjectID(id primitive.ObjectID) graphql.Marshaler {
+	return graphql.WriterFunc(func(w io.Writer) {
+		io.WriteString(w, `"`+id.Hex()+`"`)
+	})
+}
+
+// UnmarshalObjectID converts GraphQL string to MongoDB ObjectID
+func UnmarshalObjectID(v interface{}) (primitive.ObjectID, error) {
+	str, ok := v.(string)
+	if !ok {
+		return primitive.NilObjectID, fmt.Errorf("ObjectID must be a string")
+	}
+
+	objID, err := primitive.ObjectIDFromHex(str)
+	if err != nil {
+		return primitive.NilObjectID, fmt.Errorf("invalid ObjectID format")
+	}
+
+	return objID, nil
+}

pranavsuri4303 committed:
Add shared folders

pranavsuri4303 committed:
add shared folder

pranavsuri4303 committed:
add shared folder

diff --git a/equipment-service/gqlgen.yml b/equipment-service/gqlgen.yml
index 072fa8e..fec6cfd 100644
--- a/equipment-service/gqlgen.yml
+++ b/equipment-service/gqlgen.yml
@@ -102,5 +102,5 @@ models:
   DateTime:
     model: github.com/99designs/gqlgen/graphql.Time
   ObjectID:
-    model: github.com/CourtIQ/courtiq-backend/equipment-service/graph/scalar.ObjectID
+    model: github.com/CourtIQ/courtiq-backend/shared/graph/scalar.ObjectID

diff --git a/equipment-service/graph/generated.go b/equipment-service/graph/generated.go
index a2cf598..7e62bd7 100644
--- a/equipment-service/graph/generated.go
+++ b/equipment-service/graph/generated.go
@@ -17,7 +17,7 @@ import (
 	"github.com/99designs/gqlgen/graphql/introspection"
 	"github.com/99designs/gqlgen/plugin/federation/fedruntime"
 	"github.com/CourtIQ/courtiq-backend/equipment-service/graph/model"
-	"github.com/CourtIQ/courtiq-backend/equipment-service/graph/scalar"
+	"github.com/CourtIQ/courtiq-backend/shared/graph/scalar"
 	gqlparser "github.com/vektah/gqlparser/v2"
 	"github.com/vektah/gqlparser/v2/ast"
 	"go.mongodb.org/mongo-driver/bson/primitive"
diff --git a/equipment-service/graph/scalar/objectid.go b/equipment-service/graph/scalar/objectid.go
deleted file mode 100644
index 8fcaf0f..0000000
--- a/equipment-service/graph/scalar/objectid.go
+++ /dev/null
@@ -1,31 +0,0 @@
-package scalar
-
-import (
-	"fmt"
-	"io"
-
-	"github.com/99designs/gqlgen/graphql"
-	"go.mongodb.org/mongo-driver/bson/primitive"
-)
-
-// MarshalObjectID converts MongoDB ObjectID to GraphQL string
-func MarshalObjectID(id primitive.ObjectID) graphql.Marshaler {
-	return graphql.WriterFunc(func(w io.Writer) {
-		io.WriteString(w, `"`+id.Hex()+`"`)
-	})
-}
-
-// UnmarshalObjectID converts GraphQL string to MongoDB ObjectID
-func UnmarshalObjectID(v interface{}) (primitive.ObjectID, error) {
-	str, ok := v.(string)
-	if !ok {
-		return primitive.NilObjectID, fmt.Errorf("ObjectID must be a string")
-	}
-
-	objID, err := primitive.ObjectIDFromHex(str)
-	if err != nil {
-		return primitive.NilObjectID, fmt.Errorf("invalid ObjectID format")
-	}
-
-	return objID, nil
-}
diff --git a/matchup-service/graph/generated.go b/matchup-service/graph/generated.go
index 2727a31..d12170e 100644
--- a/matchup-service/graph/generated.go
+++ b/matchup-service/graph/generated.go
@@ -17,7 +17,7 @@ import (
 	"github.com/99designs/gqlgen/graphql/introspection"
 	"github.com/99designs/gqlgen/plugin/federation/fedruntime"
 	"github.com/CourtIQ/courtiq-backend/matchup-service/graph/model"
-	"github.com/CourtIQ/courtiq-backend/matchup-service/graph/scalar"
+	"github.com/CourtIQ/courtiq-backend/shared/graph/scalar"
 	gqlparser "github.com/vektah/gqlparser/v2"
 	"github.com/vektah/gqlparser/v2/ast"
 	"go.mongodb.org/mongo-driver/bson/primitive"
diff --git a/matchup-service/graph/scalar/objectid.go b/matchup-service/graph/scalar/objectid.go
deleted file mode 100644
index 8fcaf0f..0000000
--- a/matchup-service/graph/scalar/objectid.go
+++ /dev/null
@@ -1,31 +0,0 @@
-package scalar
-
-import (
-	"fmt"
-	"io"
-
-	"github.com/99designs/gqlgen/graphql"
-	"go.mongodb.org/mongo-driver/bson/primitive"
-)
-
-// MarshalObjectID converts MongoDB ObjectID to GraphQL string
-func MarshalObjectID(id primitive.ObjectID) graphql.Marshaler {
-	return graphql.WriterFunc(func(w io.Writer) {
-		io.WriteString(w, `"`+id.Hex()+`"`)
-	})
-}
-
-// UnmarshalObjectID converts GraphQL string to MongoDB ObjectID
-func UnmarshalObjectID(v interface{}) (primitive.ObjectID, error) {
-	str, ok := v.(string)
-	if !ok {
-		return primitive.NilObjectID, fmt.Errorf("ObjectID must be a string")
-	}
-
-	objID, err := primitive.ObjectIDFromHex(str)
-	if err != nil {
-		return primitive.NilObjectID, fmt.Errorf("invalid ObjectID format")
-	}
-
-	return objID, nil
-}
diff --git a/relationship-service/go.mod b/relationship-service/go.mod
index add8499..91da4b0 100644
--- a/relationship-service/go.mod
+++ b/relationship-service/go.mod
@@ -5,6 +5,7 @@ go 1.23.1
 require (
 	github.com/99designs/gqlgen v0.17.60
 	github.com/CourtIQ/courtiq-backend/equipment-service v0.0.0-20241209194043-02d71d1ba865
+	github.com/CourtIQ/courtiq-backend/shared v0.0.0-00010101000000-000000000000
 	github.com/joho/godotenv v1.5.1
 	github.com/vektah/gqlparser/v2 v2.5.20
 	go.mongodb.org/mongo-driver v1.17.1
diff --git a/relationship-service/gqlgen.yml b/relationship-service/gqlgen.yml
index 3f7c16e..8dfa62c 100644
--- a/relationship-service/gqlgen.yml
+++ b/relationship-service/gqlgen.yml
@@ -99,4 +99,4 @@ models:
   DateTime:
     model: github.com/99designs/gqlgen/graphql.Time
   ObjectID:
-    model: github.com/CourtIQ/courtiq-backend/relationship-service/graph/scalar.ObjectID
\ No newline at end of file
+    model: github.com/CourtIQ/courtiq-backend/shared/graph/scalar.ObjectID
\ No newline at end of file
diff --git a/relationship-service/graph/generated.go b/relationship-service/graph/generated.go
index 1307ecf..6743ba9 100644
--- a/relationship-service/graph/generated.go
+++ b/relationship-service/graph/generated.go
@@ -17,7 +17,7 @@ import (
 	"github.com/99designs/gqlgen/graphql/introspection"
 	"github.com/99designs/gqlgen/plugin/federation/fedruntime"
 	"github.com/CourtIQ/courtiq-backend/relationship-service/graph/model"
-	"github.com/CourtIQ/courtiq-backend/relationship-service/graph/scalar"
+	"github.com/CourtIQ/courtiq-backend/shared/graph/scalar"
 	gqlparser "github.com/vektah/gqlparser/v2"
 	"github.com/vektah/gqlparser/v2/ast"
 	"go.mongodb.org/mongo-driver/bson/primitive"
diff --git a/relationship-service/graph/scalar/objectid.go b/relationship-service/graph/scalar/objectid.go
deleted file mode 100644
index 8fcaf0f..0000000
--- a/relationship-service/graph/scalar/objectid.go
+++ /dev/null
@@ -1,31 +0,0 @@
-package scalar
-
-import (
-	"fmt"
-	"io"
-
-	"github.com/99designs/gqlgen/graphql"
-	"go.mongodb.org/mongo-driver/bson/primitive"
-)
-
-// MarshalObjectID converts MongoDB ObjectID to GraphQL string
-func MarshalObjectID(id primitive.ObjectID) graphql.Marshaler {
-	return graphql.WriterFunc(func(w io.Writer) {
-		io.WriteString(w, `"`+id.Hex()+`"`)
-	})
-}
-
-// UnmarshalObjectID converts GraphQL string to MongoDB ObjectID
-func UnmarshalObjectID(v interface{}) (primitive.ObjectID, error) {
-	str, ok := v.(string)
-	if !ok {
-		return primitive.NilObjectID, fmt.Errorf("ObjectID must be a string")
-	}
-
-	objID, err := primitive.ObjectIDFromHex(str)
-	if err != nil {
-		return primitive.NilObjectID, fmt.Errorf("invalid ObjectID format")
-	}
-
-	return objID, nil
-}
diff --git a/user-service/gqlgen.yml b/user-service/gqlgen.yml
index 612a133..a82140d 100644
--- a/user-service/gqlgen.yml
+++ b/user-service/gqlgen.yml
@@ -97,4 +97,4 @@ models:
       - github.com/99designs/gqlgen/graphql.Int64
       - github.com/99designs/gqlgen/graphql.Int32
   ObjectID:
-    model: github.com/CourtIQ/courtiq-backend/user-service/graph/scalar.ObjectID
\ No newline at end of file
+    model: github.com/CourtIQ/courtiq-backend/shared/graph/scalar.ObjectID
\ No newline at end of file
diff --git a/user-service/graph/generated.go b/user-service/graph/generated.go
index 5d016fa..cce2932 100644
--- a/user-service/graph/generated.go
+++ b/user-service/graph/generated.go
@@ -16,8 +16,8 @@ import (
 	"github.com/99designs/gqlgen/graphql"
 	"github.com/99designs/gqlgen/graphql/introspection"
 	"github.com/99designs/gqlgen/plugin/federation/fedruntime"
+	"github.com/CourtIQ/courtiq-backend/shared/graph/scalar"
 	"github.com/CourtIQ/courtiq-backend/user-service/graph/model"
-	"github.com/CourtIQ/courtiq-backend/user-service/graph/scalar"
 	gqlparser "github.com/vektah/gqlparser/v2"
 	"github.com/vektah/gqlparser/v2/ast"
 	"go.mongodb.org/mongo-driver/bson/primitive"
diff --git a/user-service/graph/scalar/objectid.go b/user-service/graph/scalar/objectid.go
deleted file mode 100644
index 8fcaf0f..0000000
--- a/user-service/graph/scalar/objectid.go
+++ /dev/null
@@ -1,31 +0,0 @@
-package scalar
-
-import (
-	"fmt"
-	"io"
-
-	"github.com/99designs/gqlgen/graphql"
-	"go.mongodb.org/mongo-driver/bson/primitive"
-)
-
-// MarshalObjectID converts MongoDB ObjectID to GraphQL string
-func MarshalObjectID(id primitive.ObjectID) graphql.Marshaler {
-	return graphql.WriterFunc(func(w io.Writer) {
-		io.WriteString(w, `"`+id.Hex()+`"`)
-	})
-}
-
-// UnmarshalObjectID converts GraphQL string to MongoDB ObjectID
-func UnmarshalObjectID(v interface{}) (primitive.ObjectID, error) {
-	str, ok := v.(string)
-	if !ok {
-		return primitive.NilObjectID, fmt.Errorf("ObjectID must be a string")
-	}
-
-	objID, err := primitive.ObjectIDFromHex(str)
-	if err != nil {
-		return primitive.NilObjectID, fmt.Errorf("invalid ObjectID format")
-	}
-
-	return objID, nil
-}

pranavsuri4303 committed:
undo shared stuff

pranavsuri4303 committed:
undo shared stuff

pranavsuri4303 committed:
Add visibility field

pranavsuri4303 committed:
Add visibility field

Commit:
Changes:

1. Implemented GetMatchUp, UpdateMatchUpStatus, AddPointToMatchUp, DeleteMatchUp;
2. Naming alignment for coherency

Commit:
WIP

Commit:
Minor text change

shaked8634 committed:
Merge branch 'main' into matchup_finishes

pranavsuri4303 committed:
Fix scalar User Service

pranavsuri4303 committed:
run gql generate with update bson tags

pranavsuri4303 commented on matchup-service/internal/services/matchup_service.go:
```diff
@@ -0,0 +1,188 @@
+	}
+
+	// Setting initial score object
+	score := model.Score{
```
need to consider how doubled would be handled. maybe need to change the Participants to [sideA: [ObjectID, ObjectID], sideB:[ObjectID, ObjectID]] This will also impact who the server is as in doubled server will switch between 4 people.

shaked8634 replied:
Replaced participants[] with participantsMap that contains list of participants on each side

pranavsuri4303 commented on matchup-service/internal/services/matchup_service.go:
```diff
@@ -0,0 +1,188 @@
+			B: participants[1],
+		},
+		CurrentScore:   &score,
+		CurrentServer:  participants[0], // TODO assuming 0
```
Same comment as above. Maybe makes sense to indentify participants by their ID and having some structure for which playing side they are one. Might need to alter the matchuUp type to accomodate.

pranavsuri4303 commented on matchup-service/internal/services/matchup_service.go:
```diff
@@ -0,0 +1,188 @@
+	return matchUp, err
+}
+
+func (s *MatchUpService) AddPointToMatchUp(ctx context.Context, matchUpFormatInput model.MatchUpFormatInput, matchUpID primitive.ObjectID) (*model.MatchUp, error) {
```
There will be instances in the match tracking where user can add a shot which does not impact the score. For example a first fault which does not impact the score but impacts the statistics  Am considering adding a mutation for the same or removing the addPoint mutation and instaed having an addShot mutation that has a referencing pointID and appends to the shots array in the point document. Not sure if it makes sense to add a shot collection.

shaked8634 replied:
When creating a point, it is possible to pass list of shotsInput object

pranavsuri4303 commented on matchup-service/internal/services/matchup_service.go:
```diff
@@ -0,0 +1,188 @@
+		NumberOfSets:   matchUpFormatInput.NumberOfSets,
+		SetFormat:      &setFormat,
+		FinalSetFormat: finalSetFormat,
+		InitialServer:  "", // TODO where does this come from?
```
should come in Matchup format. An ObjectID of the user. Need to consider doubled accomodation.

shaked8634 replied:
Done

pranavsuri4303 commented on matchup-service/internal/services/matchup_service.go:
```diff
@@ -0,0 +1,188 @@
+func (s *MatchUpService) DeleteMatchUp(ctx context.Context, matchUpID primitive.ObjectID) (*model.MatchUp, error) {
+	err := s.matchUpRepo.Delete(ctx, matchUpID)
+
+	return nil, err // TODO should this return matchup?
```
Return the deleted matchup.

pranavsuri4303 committed:
Add comments

pranavsuri4303 committed:
Add changes

diff --git a/relationship-service/graph/generated.go b/relationship-service/graph/generated.go
index 0e51912..8292e2d 100644
--- a/relationship-service/graph/generated.go
+++ b/relationship-service/graph/generated.go
@@ -789,7 +789,7 @@ func (ec *executionContext) introspectType(name string) (*introspection.Type, er
 	return introspection.WrapTypeFromDef(ec.Schema(), ec.Schema().Types[name]), nil
 }

-//go:embed "schema/enums/RelationshipStatus.gql" "schema/enums/RelationshipType.gql" "schema/interfaces/Relationship.gql" "schema/mutations/RelationshipMutations.gql" "schema/queries/RelationshipQueries.gql" "schema/scalars.gql" "schema/types/Coachship.gql" "schema/types/Friendship.gql"
+//go:embed "schema/enums/RelationshipStatus.gql" "schema/enums/RelationshipType.gql" "schema/interfaces/Relationship.gql" "schema/mutations/RelationshipMutations.gql" "schema/queries/RelationshipQueries.gql" "schema/types/Coachship.gql" "schema/types/Friendship.gql"
 var sourcesFS embed.FS

 func sourceData(filename string) string {
@@ -806,9 +806,10 @@ var sources = []*ast.Source{
 	{Name: "schema/interfaces/Relationship.gql", Input: sourceData("schema/interfaces/Relationship.gql"), BuiltIn: false},
 	{Name: "schema/mutations/RelationshipMutations.gql", Input: sourceData("schema/mutations/RelationshipMutations.gql"), BuiltIn: false},
 	{Name: "schema/queries/RelationshipQueries.gql", Input: sourceData("schema/queries/RelationshipQueries.gql"), BuiltIn: false},
-	{Name: "schema/scalars.gql", Input: sourceData("schema/scalars.gql"), BuiltIn: false},
 	{Name: "schema/types/Coachship.gql", Input: sourceData("schema/types/Coachship.gql"), BuiltIn: false},
 	{Name: "schema/types/Friendship.gql", Input: sourceData("schema/types/Friendship.gql"), BuiltIn: false},
+	{Name: "../../shared/graph/schema/Scalars.gql", Input: `scalar DateTime
+scalar ObjectID`, BuiltIn: false},
 	{Name: "../../shared/graph/schema/Visibility.gql", Input: `enum Visibility {
   PUBLIC
   PRIVATE
diff --git a/relationship-service/graph/model/models_gen.go b/relationship-service/graph/model/models_gen.go
index cf1f8ed..6cc8dfc 100644
--- a/relationship-service/graph/model/models_gen.go
+++ b/relationship-service/graph/model/models_gen.go
@@ -96,16 +96,18 @@ const (
 	RelationshipStatusPending RelationshipStatus = "PENDING"
 	// The relationship is active, meaning the request has been accepted and the relationship is established.
 	RelationshipStatusActive RelationshipStatus = "ACTIVE"
+	RelationshipStatusNone   RelationshipStatus = "NONE"
 )

 var AllRelationshipStatus = []RelationshipStatus{
 	RelationshipStatusPending,
 	RelationshipStatusActive,
+	RelationshipStatusNone,
 }

 func (e RelationshipStatus) IsValid() bool {
 	switch e {
-	case RelationshipStatusPending, RelationshipStatusActive:
+	case RelationshipStatusPending, RelationshipStatusActive, RelationshipStatusNone:
 		return true
 	}
 	return false
diff --git a/relationship-service/graph/schema/enums/RelationshipStatus.gql b/relationship-service/graph/schema/enums/RelationshipStatus.gql
index 50e81a2..745628d 100644
--- a/relationship-service/graph/schema/enums/RelationshipStatus.gql
+++ b/relationship-service/graph/schema/enums/RelationshipStatus.gql
@@ -11,4 +11,6 @@ enum RelationshipStatus {
   The relationship is active, meaning the request has been accepted and the relationship is established.
   """
   ACTIVE
+
+  NONE
 }
\ No newline at end of file
diff --git a/relationship-service/graph/schema/scalars.gql b/relationship-service/graph/schema/scalars.gql
deleted file mode 100644
index f35ecde..0000000
--- a/relationship-service/graph/schema/scalars.gql
+++ /dev/null
@@ -1,2 +0,0 @@
-scalar DateTime
-scalar ObjectID
\ No newline at end of file

pranavsuri4303 committed:
Add changes

Commit:
Further work on MatchUp service

Commit:
Merge remote-tracking branch 'origin/main' into matchup_finishes

# Conflicts:
#	matchup-service/cmd/main.go
#	matchup-service/graph/generated.go
#	matchup-service/graph/schema/scalars.gql
#	relationship-service/graph/generated.go
#	relationship-service/graph/resolvers/RelationshipQueries.resolvers.go
#	relationship-service/graph/schema/queries/RelationshipQueries.gql
#	relationship-service/internal/services/relationship_service_intf.go
#	shared/graph/schema/Scalars.gql
#	user-service/graph/generated.go
#	user-service/graph/schema/types/User.gql

Commit:
Further work on MatchUp service

Commit:
Further work on MatchUp service

shaked8634 commented:
PR updated

Commit:
Added some logic to update matchUp object

Commit:
Changes:

1. Added Game win status (GAME).
2. Assign Initial Server

Commit:
Updated missing dep
